File: tuts/tut13/heaviest.py
#!/usr/bin/python3

import sys

numbers = {}
for arg in sys.argv[1:]:
    if int(arg) not in numbers:
        numbers[int(arg)] = 1
    else:
        numbers[int(arg)] += 1

mval = list(numbers)[0]
mweight = 0

for num in numbers:
    if numbers[num]*num > mweight:
        mweight = numbers[num]*num
        mval = num

print(mval)

File: tuts/tut13/heaviest.pl
#!/usr/bin/perl


my %numbers;
my $last;
for $arg (@ARGV) {
	if (exists $numbers{$arg}) {
		$numbers{$arg} += 1;
	} else {
		$numbers{$arg} = 1;
	}

	$last = $arg;
}

$max = $numbers{$last};
$maxW = 0;
for $num (keys %numbers) {
	if ($num*$numbers{$num} > $maxW) {
		$maxW = $num*$numbers{$num};
		$max = $num;
	}
}

print "$max\n";
